dkg/sync/server.go

```diff
@@ -179,24 +190,26 @@ func (s *Server) handleStream(ctx context.Context, stream network.Stream) error
 			SyncTimestamp: msg.Timestamp,
 		}
 
-		// Verify definition hash
-		// Note: libp2p verify does another hash of defHash.
-		ok, err := pubkey.Verify(s.defHash, msg.HashSignature)
-		if err != nil {
+		if msg.Version != s.version {
+			resp.Error = errInvalidVersion
+			s.setError()
+			log.Error(ctx, "Received mismatching charon version from peer", nil,
+				z.Str("expect", s.version),
+				z.Str("got", msg.Version),
+			)
+		} else if ok, err := pubkey.Verify(s.defHash, msg.HashSignature); err != nil { // Note: libp2p verify does another hash of defHash.
 			return errors.Wrap(err, "verify sig hash")
 		} else if !ok {
 			resp.Error = errInvalidSig
-
-			s.mu.Lock()
-			s.errResponse = true
-			s.mu.Unlock()
-
+			s.setError()
```

Comment:
can we have this message verification as a separate method to server struct?

Reply:
doesn't really help much

Reply:
good idea!